torch/_inductor/fx_passes/reinplace.py

```diff
@@ -523,12 +550,18 @@ def tensor_with_same_storage_already_reinplaced(arg):
                 mutated_arg
             )
             if should_attempt_reinplace and can_inplace(node, mutated_arg):
+                # In general, we probably do not need those optimizations.
                 copy_node = copy_args_to_copy_nodes.get((mutated_arg, node))
                 if copy_node is not None:
                     replace_dict[copy_node] = copy_node.args[0]
-                for user in node.users:
-                    if user.target == operator.getitem and user.args[1] == arg:
-                        replace_dict[user] = mutated_arg
+                if not auto_functionalize:
+                    for user in node.users:
+                        # For auto_functionalize arg the index of the base, where base at index i corresponds to output at
+                        # index size(out) +i.
+                        # This used to compare string with integers before for auto_functionalize. not sure
+                        # if it was needed for inplaceable_triton_ops?
+                        if user.target == operator.getitem and user.args[1] == arg:
+                            replace_dict[user] = mutated_arg
```

Comment:
Yeah this does seem unnecessary to me, the auto_functionalized_decomp actually handles this replacement. We should try to delete this code in a follow-up

Reply:
gh issue create